core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystemImpl.java

```diff
@@ -57,7 +57,7 @@ public abstract class RelDataTypeSystemImpl implements RelDataTypeSystem {
     case INTERVAL_SECOND:
       return SqlTypeName.MAX_INTERVAL_FRACTIONAL_SECOND_PRECISION;
     default:
-      return -1;
+      return RelDataType.PRECISION_NOT_SPECIFIED;
```

Comment:
SCALE_NOT_SPECIFIED ?

Reply:
Ok, thank you for helping me review the code, I will modify it